src/Sentry/Extensibility/DisabledHub.cs

```diff
@@ -55,7 +55,7 @@ public void WithScope(Action<Scope> scopeCallback)
         /// <summary>
         /// Returns a dummy transaction.
         /// </summary>
-        public Transaction CreateTransaction(string name, string operation) => new(this);
+        public Transaction StartTransaction() => new(this);
```

Comment:
Should we have `ITransaction` and return some kind of `DisabledTransaction` here? That's how Java does this, but I personally don't think it's warranted.

Reply:
In other SDKs, Python comes to mind (/cc @untitaker ) they decided to create a transaction even if it's disabled to have a trace_id in the scope and allow errors to be linked by trace_id. But that means we're paying for the allocations and the fact instrumentations will run (since there's a transaction on the scope). 

I'm unsure what's the best way here but I lean towards the NoOp approach and if we want to correlate things we could just add the id as a tag or another approach.

Reply:
the main point of having noop transactions was that

1. you can start a child span without worrying whether you're in a transaction (the span will just be discarded)
2. you can do error correlation without enabling transaction sending

Reply:
@bruno-garcia the challenge here is that we need to decide now, otherwise changing `ITransaction` to `Transaction` is technically a breaking change (albeit I would argue highly unlikely to actually break anyone)